sources/packages/web/src/components/common/students/StudentProfile.vue

```diff
@@ -138,6 +171,53 @@ export default defineComponent({
       address.value = studentDetail.value.contact.address;
     };
 
+    const editStudentProfile = async () => {
+      await studentEditProfile.value.showModal(
+        {
+          givenNames: studentDetail.value.firstName,
+          lastName: studentDetail.value.lastName,
+          birthdate: studentDetail.value.dateOfBirth,
+          email: studentDetail.value.email,
+        },
+        saveUpdatedProfileInfo,
+      );
+    };
+
+    const saveUpdatedProfileInfo = async (
+      modalData: UpdateStudentDetailsAPIInDTO | boolean,
+    ) => {
+      try {
+        await StudentService.shared.updateStudentProfileInfo(
+          props.studentId,
+          modalData as UpdateStudentDetailsAPIInDTO,
+        );
+        await loadStudentProfile();
+        snackBar.success("Profile Information updated successfully.");
+        return true;
+      } catch (error) {
```

Comment:
Please add the below to the catch.
```ts
studentEditProfile.value.loading = false;
```

Reply:
Please correct me if I am wrong, but I don't think it is needed because it is present at the below location:

<img width="670" alt="image" src="https://github.com/user-attachments/assets/95236b77-77e4-4816-9da9-5527313a435e">

Reply:
What happens if the `beforeResolvePromisse` throws an error?
If it is reaching the line 19 it should be good then.

Reply:
If it throws an error, there is a try-catch block that catches the error, and the execution continues as usual, reaches line 19 above and sets loading to `false`.